File: Problems/Python/CS50_cash.py
import math

quaters = 0.25
dimes = 0.10
nickels = 0.05
pennies = 0.01

while True:
    change = float(input("Change: "))
    if change > 0:
        break

output = 0

while change >= quaters: 
        change = change - quaters
        output += 1
        change = round(change, 2)
        
while change >= dimes: 
        change = change % dimes
        output += 1
        change = round(change, 2)

while change >= nickels: 
        change = change % nickels
        output += 1
        change = round(change, 2)

while change >= pennies: 
        change = change % pennies
        output += 1
        change = round(change, 2)

print(output)

File: Problems/Mixed/CS50_cash.cpp
#include <iostream>
#include <string>

int main() {
    int quaters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;

    int change;
    do {std::cout << "Change owed: ";
    std::cin >> change;} while ( change < 1 );

    int cal_quaters = change / quaters;
    change %= quaters;

    int cal_dimes = change / dimes;
    change %= dimes;

    int cal_nickels = change / nickels;
    change %= nickels;

    int cal_pennies = change / pennies;
    change %= pennies;

    int output = cal_quaters + cal_pennies + cal_nickels + cal_dimes;

    std::cout << output;
}
